state/application.go

```diff
@@ -272,10 +272,6 @@ func (a *Application) removeOps(asserts bson.D) ([]txn.Op, error) {
 			Id:     a.doc.DocID,
 			Assert: asserts,
 			Remove: true,
-		}, {
-			C:      settingsC,
-			Id:     a.settingsKey(),
-			Remove: true,
```

Comment:
Yes, I do :)

1. here (propose to remove in this PR)
2. in appCharmDecRefOps
3. in finalAppCharmRemoveOps (propose to remove in this PR)

There will still be potential for duplication since we will still call finalAppCharmRemoveOps(...) inside appCharmDecRefOps(...) but it will be a lot less frequent then now.